packages/aws-rfdk/lib/core/scripts/bash/mountEfs.sh

```diff
@@ -58,6 +63,68 @@ function use_nfs_mount() {
   return $?
 }
 
+function resolve_mount_target_ip_via_api() {
+  local EFS_FS_ID=$1
+  local MNT_TARGET_RESOURCE_ID=$2
+  local AVAILABILITY_ZONE_NAME=$3
+  local AWS_REGION=$4
+  local MOUNT_POINT_IP=""
+
+  if [[ $MNT_TARGET_RESOURCE_ID == fs-* ]]; then
+    # Mounting without an access point
+    MOUNT_POINT_IP=$(aws efs describe-mount-targets \
+      --region "${AWS_REGION}"                      \
+      --file-system-id ${MNT_TARGET_RESOURCE_ID}    \
+      | jq -r ".MountTargets[] | select( .AvailabilityZoneName == \"${AVAILABILITY_ZONE_NAME}\" ) | .IpAddress" \
+    )
+  elif [[ $MNT_TARGET_RESOURCE_ID == fsap-* ]]; then
+    # Mounting via an access point
+    MOUNT_POINT_IP=$(aws efs describe-mount-targets \
+      --region "${AWS_REGION}"                      \
+      --access-point-id ${MNT_TARGET_RESOURCE_ID}   \
+      | jq -r ".MountTargets[] | select( .AvailabilityZoneName == \"${AVAILABILITY_ZONE_NAME}\" ) | .IpAddress" \
+    )
+  else
+    echo "Unsupported mount target resource: ${MNT_TARGET_RESOURCE_ID}"
+    return 1
+  fi
+
+  DNS_NAME="${EFS_FS_ID}.efs.${AWS_REGION}.amazonaws.com"
+
+  # Backup the old hosts file
+  cp /etc/hosts "/etc/hosts.rfdk-backup-$(date +%Y-%m-%dT%H:%M:%S)"
+  # Remove any existing entries for the target DNS name
+  sed -i -e "/${DNS_NAME}/d" /etc/hosts
+  # Write the resolved entry for the target DNS name
+  cat >> /etc/hosts <<EOF
+
+${MOUNT_POINT_IP} ${DNS_NAME} # Added by RFDK
+EOF
+}
+
+# Optionally resolve DNS using the EFS API
+if [[ $RESOLVE_MOUNTPOINT_IP_VIA_API == "true"  ]]
+then
+  # jq is used to query the JSON API response
+  sudo "${PACKAGE_MANAGER}" install -y jq
```

Comment:
Should check whether `jq` is present, and only hit the package manager if it is not. Use-case is an isolated machine with no network connectivity to the package repository.

Reply:
Good call! I'll add that check first